### ISSUE TITLE
Ability to import rule templates (create many rules from a single import file)

### ISSUE THREAD

**Issue by raintonr** [ioBroker/ioBroker.parser#75]
See https://github.com/UncleSamSwiss/ioBroker.loxone/issues/337#issuecomment-1270256414

Many device statistics can be read from a Loxone miniserver, which are read via HTTP requests to the device. Initially one might think reading these statistics is the job of the module specific for that device (the Loxone Miniserver) but it would be practically duplicating the work of this parser module sooooooooo...

It might be nice that if after someone spent quite some time defining all the HTTP requests necessary to fetch the full set of statistics (in the parser configuration page), that they could export that configuration for that device.

However, one would obviously need a method to select which of the items in the parser configuration they wanted to export into a template and also which of the parts of any URL are variables (in the case of Loxone, the host name and credentials for 'privileged' statistics). In this case, I think as a first step it could be cumbersome to try and create a template from within the parser admin interface and a better solution would be to focus on importing a hand crafted template file (in JSON format for example). In this way, people could create and share such files to fetch numerous items from multiple services - it clearly would not be limited to a Loxone Miniserver.

However, taking a couple of items from the Loxone Miniserver as an example, I imagine such a template file could be along the lines of (with many more entries of course):

```json
{
   "variables":[
      "host",
      "user",
      "pass"
   ],
   "items":[
      {
         "name":"$host.dev_sps_state",
         "URL":"http://$host/dev/sps/state",
         "regexp":"/value=\"(\\d*)\"/",
         "item":0,
         "role":"value",
         "type":"number",
         "interval":60000
      },
      {
         "name":"$host.dev_lan_txp",
         "URL":"http://$user:$pass@$host/dev/lan/txp",
         "regexp":"/value=\"(\\d*)\"/",
         "item":0,
         "role":"value",
         "type":"number",
         "interval":5000
      }
   ]
}
```

Variables would need to be requested at time of import, then all the parser rules would be created as a result.

**Comment by Apollon77** [comment]
In fact the adapter confog page should have a "export" button upper right, or not? If yes you coukld download the config, add the stuff you need (yes without such variables) and import that again

**Comment by raintonr** [comment]
> ... you coukld download the config, add the stuff you need (yes without such variables) and import that again

That would be go most of the way there.

But... if rules are supposed to be included in such exports that appears broken. For example, I created a rule & downloaded the config but the rule just isn't shown in there :(

**Comment by Apollon77** [comment]
Ahhh right ... the rules are stored as objects ... then your thing might be even easier ... just create objects with the correct native part

**Comment by raintonr** [comment]
Ermmmmm? Do you have an example? How would one import many objects in one go?

**Comment by Apollon77** [comment]
Check the objects created in your installations in parser.0.* ... these ARE the objects!

e.g.

```
{
  "_id": "parser.0.Donnerwetter.Knielingen.Niederschlag.Tag1",
  "common": {
    "name": "Donnerwetter.Knielingen.Niederschlag.Tag1",
    "write": false,
    "read": true,
    "unit": "l/m2",
    "type": "number",
    "role": "value"
    }
  },
  "native": {
    "link": "http://www.donnerwetter.de/agrar/region.mv?lid=XXXXX",
    "regex": "<strong>Niederschlag<\\/strong><\\/td>[^>]*\">([0-9.]+) L\\/m[^\"]+[^>]*\">[0-9.]+ L\\/m[^\"]+[^>]*\">[0-9.]+ L\\/m[^\"]+[^>]*\">[0-9.]+ L\\/m",
    "interval": "900000",
    "factor": 1,
    "offset": 0,
    "substitute": "-1",
    "comma": false,
    "item": 0,
    "substituteOld": true
  },
  "type": "state",
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  },
  "from": "system.adapter.admin.0",
  "ts": 1553291085457
}
```

**Comment by raintonr** [comment]
Yeah, so are you telling me I would have to create them one by one? That would be very painful.

**Comment by Apollon77** [comment]
You ever heared of the javascript Adapter? You can enable them to have setObject allowed ... and then use a loop... come on.

**Comment by raintonr** [comment]
Works, but not very user friendly.
